src/runtime/CMakeLists.txt

```diff
@@ -16,6 +16,10 @@ set_target_properties(leanrt PROPERTIES
 if (NOT MSVC)
 target_compile_options(leanrt PRIVATE -ftls-model=local-exec)
 endif()
+# We do not export Lean symbols when statically linking on Windows.
+if(${CMAKE_SYSTEM_NAME} MATCHES "Windows")
+target_compile_options(leanrt PRIVATE -ULEAN_EXPORTING)
+endif()
```

Comment:
@Kha I noticed that executables still exported symbols from the leanrt and leancpp. Fixing this for `leanrt` was relatively trivial (as seen above); however, after some trial and error, it is not clear to me how to fix this for `leancpp` without significant code reorganization (as `leancpp` is composed of nested object libraries and it is not possible to easily build a copy of them with different `-D` switches via CMake). What do you think we should do here? Having no exported symbols seems ideal, as it prevents the possibility of a clash when linking/loading shared libraries like `libleanshared` with the executable.

Reply:
> as it prevents the possibility of a clash when linking/loading shared libraries like `libleanshared` with the executable.

I don't understand this part, we should link against leancpp either statically or dynamically, not both

Reply:
(Resolved offline.)